electrum/tests/__init__.py

```diff
@@ -37,9 +38,12 @@ class ElectrumTestCase(SequentialTestCase):
 
     def setUp(self):
         super().setUp()
+        self.asyncio_loop, self._stop_loop, self._loop_thread = util.create_and_start_event_loop()
         self.electrum_path = tempfile.mkdtemp()
 
     def tearDown(self):
+        self.asyncio_loop.call_soon_threadsafe(self._stop_loop.set_result, 1)
```

Comment:
why is `super().tearDown()` not called first?

Reply:
Although in many cases the order does not matter, here it might, depending on the unit test framework running the tests I guess: `SequentialTestCase.setUp()` and `SequentialTestCase.tearDown()` take and release `self.test_lock`, which ensure that tests do not run in parallel. If `super().tearDown()` is called before we stop the asyncio loop, and the test framework supports running tests in parallel, some tests might fail when trying to create an "overlapping" event loop.

The natural order of calling destructors is the reverse of constructors: Create parent first, child next, but destroy child first, parent next (btw this is the order in C++, where the calls happen implicitly). Code in the child might be relying on code in the parent, so the parent shouldn't be destroyed first. I guess we could clean up the rest of the `tearDown` implementations to respect this.